lib/logstash/outputs/logstash.rb

```diff
@@ -61,129 +59,171 @@ def initialize(*a)
       fail LogStash::ConfigurationError, 'The `logstash` output does not have an externally-configurable `codec`'
     end
 
-    if @ssl_certificate_authorities && @ssl_certificate_authorities.size > 1
-      fail LogStash::ConfigurationError, 'The `logstash` output supports at most one `ssl_certificate_authorities` path'
+    @headers = {}.tap do | header |
+      header["Content-Type"] = "application/x-ndjson"
+      header["Content-Encoding"] = "gzip"
     end
```

Comment:
The use of `Hash::new` chained with `Object#tap` makes this more obscure than it needs to be.

```suggestion
    @headers = {
      "Content-Type" => "application/x-ndjson".freeze,
      "Content-Encoding" => "gzip".freeze
    }.freeze
```